napari/layers/labels/labels.py

```diff
@@ -888,6 +888,11 @@ def _get_color_lookup_func(self, data, min_label_val, max_label_val):
             self._all_vals[0] = 0
         return self._lookup_with_index
 
+    def _update_slice_response(self, response: _ImageSliceResponse) -> None:
+        """Override to convert raw slice data to displayed label colors."""
+        response = response.to_displayed(self._raw_to_displayed)
```

Comment:
why is this necessary?

Reply:
We discussed this over zoom. I think the docstring does a decent job of explaining why, but other suggestions welcome. I also considered moving the body of `to_displayed` here since this is the only place this is needed, but I will probably hold off on that for that to avoid some extra imports.

Reply:
Yeah, sorry about this. 😂 Seriously I can remember thinking, "let's just get a hacky implementation of labels going that just reuses image, we can tidy it up later." This is like Feb 2019. 😂  #3308 will slightly improve the situation but it'll still be an `astype` call...